### DISCUSSION TITLE
Max recursion occures and fixed .spec file doesn't work

### DISCUSSION THREAD

**Discussion by dsaylikoglu** [pyinstaller/pyinstaller#5756] [Help]
I wrote a python program using tkinter and sympy. However, when i try to turn it into a .exe file, pyinstaller gives me a recursion error and tells me to change something on the .spec file and try with that file. Unfortunately, when I write `import sys ; sys.setrecursionlimit(sys.getrecursionlimit() * 5)` on top of the .spec file, that also doesn't work. Please help me!! I couldn't find anything about this online. You're my last hope..

**Comment by xoviat** [comment]
add `--log-level=TRACE` and post the output to a gist.

**Comment by dsaylikoglu** [comment]
it says `'utf-8' codec can't decode byte 0x87 in position 123: invalid start byte`

**Comment by bwoodsend** [comment]
What's the command your giving to pyinstaller?

**Comment by dsaylikoglu** [comment]
`pyinstaller.exe --onefile --icon=lim.ico -w limit_calculator.spec --log-level=TRACE`

**Comment by bwoodsend** [comment]
None of those options do anything when using a spec file but that shouldn't matter. What's the contents of the `limit_calculator.spec`?

**Comment by dsaylikoglu** [comment]
```python
# -*- mode: python ; coding: utf-8 -*-
import sys ; sys.setrecursionlimit(sys.getrecursionlimit() * 5)


block_cipher = None


a = Analysis(['limit_calculator.py'],
             pathex=['C:\\Users\\lenovo\\Desktop\\Matematik\\hesap için kodlar\\exe'],
             binaries=[],
             datas=[],
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          a.binaries,
          a.zipfiles,
          a.datas,
          [],
          name='limit_calculator',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          upx_exclude=[],
          runtime_tmpdir=None,
          console=False , icon='lim.ico')
```

**Comment by bwoodsend** [comment]
> için

Looks like this really is unicode related and not just a red herring. Could you try moving your code to a folder which doesn't contain any `ç`s or any other non ascii characters? Also, what is your codepage (run `chcp` in cmd)?

**Comment by dsaylikoglu** [comment]
My active code page is 437.

**Comment by xoviat** [comment]
first of all, what is the stack trace for the error `'utf-8' codec can't decode byte 0x87 in position 123: invalid start byte`?

**Comment by bwoodsend** [comment]
That ç is the culprit.

```python
>>> "ç".encode("cp437")
b'\x87'
>>> "ç".encode("cp437").decode("utf-8")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x87 in position 0: invalid start byte
```

I suspect that whatever text editor you use has converted the spec file to your native encoding instead of leaving it as `utf-8` as at should be. You'll have to work out how to change it back which depends on which text editor you're using.

**Comment by dsaylikoglu** [comment]
i changed the unicodes chars to ascii characters. still not happening. i ran the command in a pc and it works. i guess my pc has sth wrong ones again

**Comment by bwoodsend** [comment]
Full traceback please.

**Comment by dsaylikoglu** [comment]
```
10228 INFO: PyInstaller: 5.0.dev0
10229 INFO: Python: 3.7.6 (conda)
10230 INFO: Platform: Windows-10-10.0.18362-SP0
10233 INFO: wrote C:\Users\lenovo\Desktop\Matematik\hesap icin kodlar\exe\limit_calculator.spec
10256 INFO: UPX is not available.
10270 INFO: Extending PYTHONPATH with paths
['C:\\Users\\lenovo\\Desktop\\Matematik\\hesap icin kodlar\\exe',
 'C:\\Users\\lenovo\\Desktop\\Matematik\\hesap icin kodlar\\exe']
11692 INFO: checking Analysis
11693 INFO: Building Analysis because Analysis-00.toc is non existent
11694 INFO: Initializing module dependency graph...
11715 INFO: Caching module graph hooks...
11783 INFO: Analyzing base_library.zip ...
36520 INFO: Caching module dependency graph...
37245 INFO: running Analysis Analysis-00.toc
37308 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\users\lenovo\anaconda\python.exe
38290 INFO: Analyzing C:\Users\lenovo\Desktop\Matematik\hesap icin kodlar\exe\limit_calculator.py
51957 INFO: Processing pre-find module path hook distutils from 'c:\\users\\lenovo\\anaconda\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-distutils.py'.
51962 INFO: distutils: retargeting to non-venv dir 'c:\\users\\lenovo\\anaconda\\lib'
53643 INFO: Processing pre-find module path hook site from 'c:\\users\\lenovo\\anaconda\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-site.py'.
53651 INFO: site: retargeting to fake-dir 'c:\\users\\lenovo\\anaconda\\lib\\site-packages\\PyInstaller\\fake-modules'
153281 INFO: Processing pre-safe import module hook urllib3.packages.six.moves from 'c:\\users\\lenovo\\anaconda\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-urllib3.packages.six.moves.py'.
174714 INFO: Processing pre-safe import module hook six.moves from 'c:\\users\\lenovo\\anaconda\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-six.moves.py'.

=============================================================
A RecursionError (maximum recursion depth exceeded) occurred.
For working around please follow these instructions
=============================================================

1. In your program's .spec file add this line near the top::

     import sys ; sys.setrecursionlimit(sys.getrecursionlimit() * 5)

2. Build your program by running PyInstaller with the .spec file as
   argument::

     pyinstaller myprog.spec

3. If this fails, you most probably hit an endless recursion in
   PyInstaller. Please try to track this down has far as possible,
   create a minimal example so we can reproduce and open an issue at
   https://github.com/pyinstaller/pyinstaller/issues following the
   instructions in the issue template. Many thanks.

Explanation: Python's stack-limit is a safety-belt against endless recursion,
eating up memory. PyInstaller imports modules recursively. If the structure
how modules are imported within your program is awkward, this leads to the
nesting being too deep and hitting Python's stack-limit.

With the default recursion limit (1000), the recursion error occurs at about
115 nested imported, with limit 2000 at about 240, with limit 5000 at about
660.
```

**Comment by xoviat** [comment]
You need to add `--log-level=TRACE` to the PyInstaller command. And if you run into unicode errors, you need to fix them.

**Comment by xoviat** [comment]
If `--log-level=TRACE` is not working, then the solution is to find out why. Not ignore it.

**Comment by bwoodsend** [comment]
@dsaylikoglu That's not the unicode error. What was the traceback from that one?

It looks like your using full blown Anaconda (i.e. not MiniConda) which means that your environment will be huge because Anaconda installs hundreds of huge packages. That's why you keep getting recursion errors. The fix is to create a virtual environment containing PyInstaller and the packages you need. It's up to you whether you use venv or conda to do this.

**Comment by dsaylikoglu** [comment]
i tried with a conda env but didn't work

**Comment by bwoodsend** [comment]
> but didn't work

Could you be a little more descriptive?

**Comment by dsaylikoglu** [comment]
I created a conda env inside the project's folder and tried to run the cmd there, but as I said, it failed.

**Comment by bwoodsend** [comment]
What commands did you use to create the environment, activate the environment, run PyInstaller, test your application? Which step failed and what was the full error message?

**Comment by Anonteam798** [comment]
Hello guys, I have the same issue with pyinstaller and I cant find any solution

**Comment by bwoodsend** [comment]
[sigh]. @Anonteam798 Can you please fill out the [issue template](https://github.com/pyinstaller/pyinstaller/blob/develop/.github/ISSUE_TEMPLATE/bug_report.md) so we know what your problem is.

**Comment by Anonteam798** [comment]
I can't convert my programm from ".py" to ".exe" with pyinstaller, it shows me a recursive error, and try to write in ".spec", then I TRY again to convert but when it finish, I can't run my programm ".exe" from the directory, 

I have python 3.7, My programm is a GUI with TKINTER 
 What can I do to solve my problem? Please help me

**Comment by rokm** [comment]
> but when it finish

If it finished, then you're not having recursion error anymore.

> I can't run my programm ".exe" from the directory,

What's preventing you from doing that?

**Reply by Anonteam798** [reply]
Hi bro, I solve my problem, creating a virtual environment with ANACONDA , install the main packages, and finally it worked, but  I'm  confused because with pyinstaller I COMPILE one python file but my programm depends on another python file 
What can I do?
